crates/sovereign-sdk/full-node/db/sov-db/src/ledger_db/traits.rs

```diff
@@ -40,9 +40,10 @@ pub trait SharedLedgerOps {
     ) -> Result<()>;
 
     /// Commits a soft confirmation to the database by inserting its transactions and batches before
-    fn commit_soft_confirmation<B: Serialize, T: Serialize, DS: DaSpec>(
+    fn commit_soft_confirmation<T: Serialize, DS: DaSpec>(
         &self,
-        batch_receipt: SoftConfirmationReceipt<B, T, DS>,
+        state_root: &[u8],
```

Comment:
state_root is [u8; 32]?

Reply:
well deep down it was written to db as vec

and I only had access to as_ref :D

Reply:
That's unfortunate. It would be nice to pass [u8; 32] =)